kyo-core/shared/src/main/scala/kyo/Console.scala

```diff
@@ -49,22 +55,19 @@ object Console:
 
     /** A live implementation of the Console trait.
       */
-    val live: Console =
-        Console(
-            new Unsafe:
-                def readln()(using AllowUnsafe) =
-                    val line = scala.Console.in.readLine()
-                    if line == null then
-                        throw new EOFException("Consoles.readln failed.")
-                    else
-                        line
-                    end if
-                end readln
-                def print(s: String)(using AllowUnsafe)      = scala.Console.out.print(s)
-                def printErr(s: String)(using AllowUnsafe)   = scala.Console.err.print(s)
-                def println(s: String)(using AllowUnsafe)    = scala.Console.out.println(s)
-                def printlnErr(s: String)(using AllowUnsafe) = scala.Console.err.println(s)
-        )
+    val live: Console = Console(
+        new Unsafe:
+            def readln()(using AllowUnsafe) =
+                Result.catching(Maybe(scala.Console.in.readLine())).flatMap {
+                    case Absent         => Result.fail(new EOFException("Consoles.readln failed."))
+                    case Present(value) => Result.success(value)
+                }
+            def print(s: String)(using AllowUnsafe)      = Result.catching(scala.Console.out.print(s))
```

Comment:
Will this catch only the type in the signature?

Reply:
Yes, the exception type is inferred but I've added explicit type params to avoid ambiguity